src/main/java/com/minecolonies/coremod/quests/objectives/DialogueObjectiveTemplateTemplate.java

```diff
@@ -107,6 +109,87 @@ public IObjectiveInstance startObjective(final IQuestInstance colonyQuest)
         {
             colonyQuest.getParticipant(target).openDialogue(colonyQuest, colonyQuest.getIndex());
         }
-        return null;
+        return createObjectiveInstance();
+    }
+
+    @Override
+    public @NotNull IObjectiveInstance createObjectiveInstance()
+    {
+        return new DialogueProgressInstance(this);
+    }
+
+    /**
+     * Progress data of this objective.
+     */
+    public static class DialogueProgressInstance implements IDialogueObjectInstance
+    {
+        /**
+         * Tag for has interacted the NBT state of this progress instance.
+         */
+        private static final String TAG_HAS_INTERACTED = "hasInteracted";
+
+        /**
+         * The template belonging to this progress instance.
+         */
+        private final DialogueObjectiveTemplateTemplate template;
+
+        /**
+         * Whether the player has completed the dialogue interaction.
+         */
+        private boolean hasInteracted = false;
+
+        /**
+         * Default constructor.
+         */
+        public DialogueProgressInstance(final DialogueObjectiveTemplateTemplate template)
+        {
+            this.template = template;
+        }
+
+        @Override
+        public boolean isFulfilled()
+        {
+            return hasInteracted;
+        }
+
+        @Override
+        public int getMissingQuantity()
+        {
+            return hasInteracted ? 0 : 1;
+        }
+
+        @Override
+        public Component getProgressText(final IQuestInstance quest, final Style style)
```

Comment:
This text could be on the objective template or instance, doesnt have to be in here then

Reply:
It can't because it relies on active progress data from the quest objective.
For example for killing entities I need to know how many were killed, that data only lives within the objective instance for that quest.

Reply:
The reason I didn't do that is because it'd only be able to accept a generic interface, which would mean that each objective template first would have to cast it to the correct instance to see if the objective instance even belongs there.

If I have the method directly in the objective instance itself, I know I have the correct classes already.